src/main/java/net/javadiscord/javabot/api/routes/metrics/MetricsController.java

```diff
@@ -0,0 +1,63 @@
+package net.javadiscord.javabot.api.routes.metrics;
+
+import com.github.benmanes.caffeine.cache.Caffeine;
+import lombok.extern.slf4j.Slf4j;
+import net.dv8tion.jda.api.entities.Guild;
+import net.javadiscord.javabot.Bot;
+import net.javadiscord.javabot.api.response.ApiResponseBuilder;
+import net.javadiscord.javabot.api.response.ApiResponses;
+import net.javadiscord.javabot.api.routes.CaffeineCache;
+import net.javadiscord.javabot.api.routes.JDAEntity;
+import net.javadiscord.javabot.api.routes.metrics.model.MetricsData;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.RestController;
+
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Handles all GET-Requests on the {guild_id}/metrics route.
+ */
+@Slf4j
+@RestController
+public class MetricsController extends CaffeineCache<Long, MetricsData> implements JDAEntity {
+
+	/**
+	 * The constructor of this class which initializes the {@link Caffeine} cache.
+	 */
+	public MetricsController() {
+		super(Caffeine.newBuilder()
+				.expireAfterWrite(15, TimeUnit.MINUTES)
+				.build()
+		);
+	}
+
+	/**
+	 * Serves metrics for the specified guild.
+	 *
+	 * @param guildId The guilds' id.
+	 * @return The {@link ResponseEntity}.
+	 */
+	@GetMapping(
+			value = "{guild_id}/metrics",
+			produces = MediaType.APPLICATION_JSON_VALUE
+	)
+	public ResponseEntity<String> getMetrics(@PathVariable(value = "guild_id") String guildId) {
```

Comment:
Spring Web already uses Jackson and is able to serialize objects with that when using `@RestController`.
If you just return a `MetricsData`-object (`ResponseEntity<MetricsData>` or `ResponseEntity<?>` if it may be one of multiple types should work as well), Spring will try to serialize that using Jackson.
With that, `produces = MediaType.APPLICATION_JSON_VALUE` might be obsolete.